zvariant/src/dict.rs

```diff
@@ -160,6 +168,33 @@ impl Display for Dict<'_, '_> {
     }
 }
 
+impl<'a, 'k, 'v> IntoIterator for &'a Dict<'k, 'v> {
+    type Item = (&'a Value<'k>, &'a Value<'v>);
+    type IntoIter = <&'a BTreeMap<Value<'k>, Value<'v>> as IntoIterator>::IntoIter;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.map.iter()
+    }
+}
+
+impl<'a, 'k, 'v> IntoIterator for &'a mut Dict<'k, 'v> {
+    type Item = (&'a Value<'k>, &'a mut Value<'v>);
+    type IntoIter = <&'a mut BTreeMap<Value<'k>, Value<'v>> as IntoIterator>::IntoIter;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.map.iter_mut()
+    }
+}
+
+impl<'k, 'v> IntoIterator for Dict<'k, 'v> {
+    type Item = (Value<'k>, Value<'v>);
+    type IntoIter = <BTreeMap<Value<'k>, Value<'v>> as IntoIterator>::IntoIter;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.map.into_iter()
+    }
+}
+
```

Comment:
I'm not sure we really need these. I doubt there will be enough users of this to justify this code, just to help avoid `.iter()` call. Besides, it a bit strange to implement `IntoIterator` if we don't implement `Iterator`. I don't think any type does that (at least in std).

Reply:
I suggest that we should keep these because it improves the completeness of `Iterator` related code for `zvariant::Dict` type.

To clarify the statement of
> I doubt there will be enough users of this to justify this code, just to help avoid `.iter()` call.
It might seem so for all reference related types which in current case are `&Dict` and `&mut Dict`. However, `IntoIterator` for `Dict` is important which consumes the variable and allows the user to avoid calling `.clone()` for each value especially when the `zvariant::Value` is an `OwnedValue`.

To further clarify the completeness of `IntoIterator` trait implementation for reference types and value types, I would suggest the sample implementation for `HashMap` in standard library.
https://github.com/rust-lang/rust/blob/1.78.0/library/std/src/collections/hash/map.rs#L2175

Reply:
> However, `IntoIterator` for `Dict` is important which consumes the variable and allows the user to avoid calling `.clone()` for each value especially when the `zvariant::Value` is an `OwnedValue`.

Sure, that's a good point. Let's keep the last `IntoIter` impl then and remove the first 2 (the reference ones).

Reply:
Removed the first 2.